File: LibraryApp/src/models/book.ts
export interface Book {
    id: number;
    title: string;
    author: string;
    description: string;
    coverImage: string | null;
    publisher: string;
    publicationDate: string;
    category: string;
    isbn: string;
    pageCount: number;
    isCheckedOut: boolean;
    returnDueDate: string | null;
}

File: LibraryApi/LibraryApi/DTOs/Book.cs
using LibraryApi.Data;

namespace LibraryApi.DTOs;

public class Book
{
    public int Id { get; set; }
    public string Title { get; set; } = string.Empty;
    public string Author { get; set; } = string.Empty;
    public string Description { get; set; } = string.Empty;
    public byte[]? CoverImage { get; set; }
    public string Publisher { get; set; } = string.Empty;
    public DateTime PublicationDate { get; set; }
    public string Category { get; set; } = string.Empty;
    public string Isbn { get; set; } = string.Empty;
    public int PageCount { get; set; }
    public bool IsCheckedOut { get; set; }
    public DateTime? ReturnDueDate { get; set; }

}
